Update module documentation about ref links (#76088)
ansible/ansible: docs/docsite/rst/dev_guide/developing_modules_documenting.rst

        - ref: aci_guide
          description: Detailed information on how to manage your ACI infrastructure using Ansible.

        # Reference by rST documentation anchor (with custom title)
        - ref: The official Ansible ACI guide <aci_guide>
          description: Detailed information on how to manage your ACI infrastructure using Ansible.

        # Reference by Internet resource
        - name: APIC Management Information Model reference
          description: Complete reference of the APIC object model.
          link: https://developer.cisco.com/docs/apic-mim-ref/

>   * Using ``ref:`` with a reference that is not published to the public docsite may require the use of a title to work correctly.

Reviewer: ```suggestion
  * If you use ``ref:`` to link to a page that does not exist on docs.ansible.com (for example, documentation for a module you are adding on your branch), you might need to add a title to make the link work correctly.
```

>   * If you use ``ref:`` to link to an anchor that isn't associated with a title, you must add a title to the ref for the link to work correctly.

Reviewer: ```suggestion
  * If you use ``ref:`` to link to an anchor that not associated with a title, you must add a title to the ref for the link to work correctly.
```

Author: ```suggestion
  * If you use ``ref:`` to link to an anchor that is not associated with a title, you must add a title to the ref for the link to work correctly.
```
typo fix from above suggestion

>   * If you use ``ref:`` to link to a page that does not exist on docs.ansible.com (for example, documentation for a module you are adding on your branch), you might need to add a title to make the link work correctly.

Reviewer 2: Are you sure that this is only the case for modules (or plugins) that do not exist on docs.ansible.com? I would think it's the case for *all* modules and plugins.

Author: well.. this is the discussion I was raising about whether it should be considered a bug or not. I kind of feel like it should be. Because if the content is published, then it will "work" in the sense that you will get a rendered link and no warnings or errors. The problem is that the link will always point to the public docsite; which again imo is incorrect.

OTOH, every current use of `ref:` that I could find has no title. So in fact we may want to correct all of them as well.

If this is a bug, I'm not sure where to file it either, because I don't know where that behavior comes from.

Author: Based on our conversation in IRC, I've updated the wording.

>   * Non-module plugins can be linked to with ``ref:`` using the rST anchor, for example ``ref: The documentation for namespace.collection.plugin_name lookup <ansible_collections.namespace.collection.plugin_name_lookup>``.

Reviewer: ```suggestion
  * You can link to non-module plugins with ``ref:`` using the rST anchor, for example ``ref: The documentation for namespace.collection.plugin_name lookup <ansible_collections.namespace.collection.plugin_name_lookup>``.
```

>   * You can link to non-module plugins with ``ref:`` using the rST anchor, but note that plugin and module anchors are never associated with a title, so a title must always be supplied. For example ``ref: namespace.collection.plugin_name lookup plugin  <ansible_collections.namespace.collection.plugin_name_lookup>``.

Reviewer: ```suggestion
  * You can link to non-module plugins with ``ref:`` using the rST anchor, but plugin and module anchors are never associated with a title, so you must supply a title when you link to them. For example ``ref: namespace.collection.plugin_name lookup plugin  <ansible_collections.namespace.collection.plugin_name_lookup>``.
```

>   * You can link to non-module plugins with ``ref:`` using the rST anchor, but note that plugin and module anchors are never associated with a title, so a title must always be supplied. For example ``ref: The documentation for namespace.collection.plugin_name lookup <ansible_collections.namespace.collection.plugin_name_lookup>``.

Reviewer 2: If it should be more similar to how module references work, it should be:
```suggestion
  * You can link to non-module plugins with ``ref:`` using the rST anchor, but note that plugin and module anchors are never associated with a title, so a title must always be supplied. For example ``ref: namespace.collection.plugin_name  <ansible_collections.namespace.collection.plugin_name_lookup>``.
```
with an appropriate description which mentions the plugin type. Though it's probably a good idea to include the plugin type:
```suggestion
  * You can link to non-module plugins with ``ref:`` using the rST anchor, but note that plugin and module anchors are never associated with a title, so a title must always be supplied. For example ``ref: namespace.collection.plugin_name lookup plugin  <ansible_collections.namespace.collection.plugin_name_lookup>``.
```

Author: Right, not including the plugin type in the title can (and will soon in the case of my collection) end up with duplicate titles because of modules and plugins with the same name.
